src/OpenFeature/ProviderRepository.cs

```diff
@@ -44,31 +44,18 @@ public async ValueTask DisposeAsync()
         /// </summary>
         /// <param name="featureProvider">the provider to set as the default, passing null has no effect</param>
         /// <param name="context">the context to initialize the provider with</param>
-        /// <param name="afterSet">
-        /// <para>
-        /// Called after the provider is set, but before any actions are taken on it.
-        /// </para>
-        /// This can be used for tasks such as registering event handlers. It should be noted that this can be called
-        /// several times for a single provider. For instance registering a provider with multiple names or as the
-        /// default and named provider.
-        /// <para>
-        /// </para>
-        /// </param>
-        /// <param name="afterInitialization">
+        /// <param name="afterInitSuccess">
         /// called after the provider has initialized successfully, only called if the provider needed initialization
         /// </param>
-        /// <param name="afterError">
+        /// <param name="afterInitError">
         /// called if an error happens during the initialization of the provider, only called if the provider needed
         /// initialization
         /// </param>
-        /// <param name="afterShutdown">called after a provider is shutdown, can be used to remove event handlers</param>
         public async Task SetProvider(
             FeatureProvider? featureProvider,
             EvaluationContext context,
-            Action<FeatureProvider>? afterSet = null,
-            Action<FeatureProvider>? afterInitialization = null,
-            Action<FeatureProvider, Exception>? afterError = null,
-            Action<FeatureProvider>? afterShutdown = null)
```

Comment:
`afterSet` and `afterShutdown` were only used in tests. Removing them for simplicity.